sample/src/main/java/dev/arkbuilders/sample/ScoreActivity.kt

```diff
@@ -0,0 +1,119 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.button.MaterialButton
+import dev.arkbuilders.arklib.data.folders.FoldersRepo
+import dev.arkbuilders.arklib.data.index.ResourceIndex
+import dev.arkbuilders.arklib.data.index.ResourceIndexRepo
+import dev.arkbuilders.arklib.user.score.ScoreStorage
+import dev.arkbuilders.arklib.user.score.ScoreStorageRepo
+import dev.arkbuilders.components.filepicker.ArkFilePickerConfig
+import dev.arkbuilders.components.filepicker.ArkFilePickerFragment
+import dev.arkbuilders.components.filepicker.ArkFilePickerMode
+import dev.arkbuilders.components.filepicker.onArkPathPicked
+import dev.arkbuilders.components.scorewidget.HorizontalScoreWidgetComposable
+import dev.arkbuilders.components.scorewidget.ScoreWidgetController
+import dev.arkbuilders.components.scorewidget.VerticalScoreWidgetComposable
+import kotlinx.coroutines.launch
+import java.nio.file.Path
+import kotlin.io.path.name
+
+class ScoreActivity : AppCompatActivity() {
+    private var rootFolder: Path? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_score)
+
+        val btnPickRoot = findViewById<MaterialButton>(R.id.btn_pick_root)
+        val btnPickResource = findViewById<MaterialButton>(R.id.btn_pick_resource)
+
+        supportFragmentManager.onArkPathPicked(
+            this,
+            customRequestKey = PICK_ROOT_KEY
+        ) { root ->
+            rootFolder = root
+            btnPickRoot.text = root.name
+        }
+
+        supportFragmentManager.onArkPathPicked(
+            this,
+            customRequestKey = PICK_RESOURCE_KEY
+        ) { resourcePath ->
+            btnPickResource.text = resourcePath.name
+            rootFolder?.let {
+                onResourcePicked(root = it, resourcePath)
+            }
+        }
+
+        btnPickRoot.setOnClickListener {
+            ArkFilePickerFragment
+                .newInstance(ArkFilePickerConfig(pathPickedRequestKey = PICK_ROOT_KEY))
+                .show(supportFragmentManager, null)
+        }
+
+        btnPickResource.setOnClickListener {
+            ArkFilePickerFragment
+                .newInstance(
+                    ArkFilePickerConfig(
+                        pathPickedRequestKey = PICK_RESOURCE_KEY,
+                        mode = ArkFilePickerMode.FILE
+                    )
+                ).show(supportFragmentManager, null)
+        }
+    }
+
+    private fun onResourcePicked(
+        root: Path,
+        resourcePath: Path
+    ) = lifecycleScope.launch {
+        val (index, scoreStorage) = setupIndexAndScoreStorage(root)
+        val id = index.allPaths().toList()
+            .find { it.second == resourcePath }!!.first
```

Comment:
The result of `find` could be `null`, and app will crash in that case because of `!!` assertion.

Reply:
Fixed. This can happen if file is selected from outside root